challenges/sre/challenge-security.md

```diff
@@ -0,0 +1,202 @@
+
+### Suggestions
+* Complete the design document within the first week.  This is to ensure you have enough time to work on implementation.
+* Avoid writing an overly detailed design document. 500-1500 words is a good target.
+* We encourage you to ask questions in Slack.
+* Avoid sending draft PRs for feedback as it is difficult to evaluate which parts are draft or complete.
+
+## Implementation
+
+### Server
+Once the design document is approved, the next PR should include your server code, along with build automation to simplify testing.  Use the [redis-cli](https://redis.io/learn/guides/import) to import the [test data](files/data.redis) provided as this is needed for the final solution. The interview panel will clone the repository and run Make targets to build and test your server code. Be sure to include a couple of high-quality tests that cover both happy and unhappy scenarios, and minimize external dependencies.
+
+### Service Levels & Deployment
+
+After the server code is approved, submit a final PR including service levels, deployment, and any remaining requirements.
+
+Your chosen service levels should focus on meaningful indicators, be set at realistic thresholds, and align with the system's functionality and user experience.  The presentation of these SLOs is up to the candidate's discretion.  Consider a simple solution and include comments on how a more robust solution could be implemented (see [trade-offs](#trade-offs)).
+
+When selecting [security invariants](https://d1.awsstatic.com/events/aws-reinforce-2022/NIS234_Learn-to-create-end-to-end-security-invariants-using-AWS-services.pdf), consider which properties you expect your system to always maintain.  The chosen invariants should cover attack vectors against the most critical parts of your system design. How these properties are evaluated and reported is up to the candidate.
+
+This part of the implementation will be evaluated ensuring the helm chart includes production-level packaging, service to service communication is secured implementing zero-trust, all artifacts are easy to build and deploy, and that the selected security invariants of the system can be evaluated.
+
+### Tooling
+For evaluation purposes, your solution should be written in Go and is deployable to a local Kubernetes cluster. The choice of which local Kubernetes cluster is up to you, but please ensure compatibility with both macOS and Linux. We suggest [KIND](https://kind.sigs.k8s.io/).
+
+You may use additional external dependencies, but ensure that detecting or installing these are straightforward for the reviewer. At a minimum, your solution should include the following dependencies:
+
+* Go
+* make
+* Docker
+
+## Level 4
+
+### Server
+* HTTPS API to create, get, or list users with Redis datastore
+* HTTPS API to add a new payment with Redis datastore
+* HTTPS API to get payments by user with Redis datastore
+* HTTPS Health check ensuring availability
+* Create a secure client API with mTLS
```

Comment:
What is this? A Go client?

Reply:
```suggestion
* Secure connections between the HTTPS API and caller with mTLS
```

I might revert the wording here to match the previous challenge since that is the aim